mongodb.py

```diff
@@ -16,24 +16,29 @@ def __init__(self):
         self.mongo_db = ["admin", ]
         self.mongo_user = None
         self.mongo_password = None
-        self.mongo_monitoring_level = None
+        self.mongo_version = None
+        self.cluster_name = None
 
+        self.plugin_instance = None
         self.lockTotalTime = None
         self.lockTime = None
         self.accesses = None
         self.misses = None
 
-    def submit(self, type, instance, value, db=None):
-        if db:
-            plugin_instance = '%s-%s' % (self.mongo_port, db)
-        else:
-            plugin_instance = str(self.mongo_port)
+    def submit(self, type, type_instance, value, db=None):
         v = collectd.Values()
         v.plugin = self.plugin_name
-        v.plugin_instance = plugin_instance
+
+        if self.cluster_name is not None:
```

Comment:
Keep previous logic for determining base plugin_instance name
Then add [cluster=<cluster_name>] if cluster is set.